src/Orc.Metadata/Providers/FastMemberInvokerMetadataProvider.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace Orc.Metadata.Providers
+{
+    using System;
+    using System.Threading.Tasks;
+    using Catel.Caching;
+    using Catel.Reflection;
+    using Catel.Threading;
+
+    public class FastMemberInvokerMetadataProvider : IMetadataProvider
+    {
+        private static readonly ICacheStorage<Type, IFastMemberInvoker> MemberInvokerCache = new CacheStorage<Type, IFastMemberInvoker>();
+
+        public Task<IObjectWithMetadata> GetMetadataAsync(object obj)
+        {
+            var memberInvoker = GetMemberInvoker(obj);
+
+            return TaskHelper<IObjectWithMetadata>.FromResult(new FastMemberInvokerObjectWithMetadata(obj, memberInvoker));
+        }
+
+        private IFastMemberInvoker GetMemberInvoker(object obj)
+        {
+            var invokerType = typeof(FastMemberInvoker<>).MakeGenericType(obj.GetType());
```

Comment:
Cache this. Reflection is really expensive, so cache the result of MakeGenericType in Dictionary<Type, IFastMemberInvoker)